compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -87,71 +71,46 @@ public fun <T : Component> SwingPanel(
     update: (T) -> Unit = NoOpUpdate,
 ) {
     val interopContainer = LocalInteropContainer.current
-    val compositeKey = currentCompositeKeyHash
-    val interopViewHolder = remember {
-        SwingInteropViewHolder2(
-            container = interopContainer,
-            group = SwingInteropViewGroup(
-                key = compositeKey,
-                focusComponent = interopContainer.root,
-            ),
-            update = update,
-        )
-    }
-
-    val density = LocalDensity.current
+    val compositeKeyHash = currentCompositeKeyHash
     val focusManager = LocalFocusManager.current
-    val focusSwitcher = remember { FocusSwitcher(interopViewHolder, focusManager) }
 
-    OverlayLayout(
-        modifier = modifier.onGloballyPositioned { coordinates ->
-            val rootCoordinates = coordinates.findRootCoordinates()
-            val clippedBounds = rootCoordinates
-                .localBoundingBoxOf(coordinates, clipBounds = true).round(density)
-            val bounds = rootCoordinates
-                .localBoundingBoxOf(coordinates, clipBounds = false).round(density)
+    // TODO: entire interop context must be inside SwingInteropViewHolder in order to
+    //  expose a version of this API with `onReset` callback and integrated with ReusableComposeNode
+    //  https://youtrack.jetbrains.com/issue/CMP-5897/Desktop-self-contained-InteropViewHolder
 
-            interopViewHolder.setBounds(bounds, clippedBounds)
-        }.drawBehind {
-            // Clear interop area to make visible the component under our canvas.
-            drawRect(Color.Transparent, blendMode = BlendMode.Clear)
-        }.trackInteropPlacement(interopViewHolder)
-            .pointerInteropFilter(interopViewHolder)
-    ) {
-        focusSwitcher.Content()
+    val group = remember {
```

Comment:
Otherwise it will be recreated on every recomposition, even if not used, apparently?

Reply:
I mean it can be moved to `SwingInteropViewHolder` remember block

Reply:
It will require `FocusSwitcher.group` to be internal instead of private. That kinda ruins the logic here a little bit.
`Holder` depends on `group` and `focusSwitcher`. `focusSwitcher` depends on `group`, but it's an implementation detail and can be likely removed in future iteration.